### PR TITLE
tests: use nyc for coverage, dropping deprecated istanbul

### PR DESCRIPTION
**Pull request by paulirish**
Istanbul has been choking on async/await since that landed in master.
This fixes that.

### CODE REVIEW

**Comment by paulirish** [comment]
yah! np. :)

check this!
image

looking into it.. 

this stands out.. i should make sure the test coverage of test files isn't calculated..
https://codecov.io/gh/GoogleChrome/lighthouse/tree/f71fe0f89487c215d17d435e9b44c733a0424952/lighthouse-core/test

**Comment by paulirish** [comment]
to confirm, this is ready for a new review. I've looked over the coverage data and am happy with where things stand.

File: package.json [hunk 1]
```diff
@@ -24,23 +24,27 @@
...
+    "unit-core": "mocha --reporter dot \"lighthouse-core/test/**/*-test.js\"",
+    "unit-cli": "mocha --reporter dot \"lighthouse-cli/test/**/*-test.js\"",
+    "unit-viewer": "mocha --reporter dot \"lighthouse-viewer/test/**/*-test.js\"",
+    "unit": "yarn unit-core && yarn unit-cli && yarn unit-viewer",
     "core-unit": "yarn unit-core",
     "cli-unit": "yarn unit-cli",
     "viewer-unit": "yarn unit-viewer",
+    "watch": "yarn unit-core --watch",
+
+    "unit:silentcoverage": "nyc --silent yarn unit",
```
Comment (patrickhulce) [inline package.json]: why silent coverage? just got annoyed with it in output?

File: lighthouse-core/scripts/run-mocha.sh [hunk 2]
```diff
@@ -1,27 +0,0 @@
-#!/usr/bin/env bash
```
Comment (patrickhulce) [inline lighthouse-core/scripts/run-mocha.sh]: 🎉

File: package.json [hunk 3]
```diff
@@ -24,23 +24,27 @@
...
     "debug": "node --inspect-brk ./lighthouse-cli/index.js",
     "start": "node ./lighthouse-cli/index.js",
     "test": "yarn lint --quiet && yarn unit && yarn type-check && yarn closure",
     "test-extension": "cd lighthouse-extension && yarn test",
-    "unit-core": "bash lighthouse-core/scripts/run-mocha.sh --core",
-    "unit-cli": "bash lighthouse-core/scripts/run-mocha.sh --cli",
-    "unit-viewer": "bash lighthouse-core/scripts/run-mocha.sh --viewer",
-    "unit": "bash lighthouse-core/scripts/run-mocha.sh --default",
+
+    "unit-core": "mocha --reporter dot \"lighthouse-core/test/**/*-test.js\"",
```
Comment (patrickhulce) [inline package.json]: nit: can we just use single quotes instead? escaping hurts my 👀

File: package.json (hunk 1 above)
Comment (paulirish) [reply package.json]: yeah here's an example: https://travis-ci.org/GoogleChrome/lighthouse/builds/361959444?utm_source=github_status&utm_medium=notification

**Pull request merged**